quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnamePropertyMappers.java

```diff
@@ -44,6 +53,17 @@ private HostnamePropertyMappers(){}
         };
     }
 
+    private static String getValidatedLowercaseHostname(String value, ConfigSourceInterceptorContext configSourceInterceptorContext) {
+
+        String hostnameVal = value.toLowerCase(Locale.ROOT);
```

Comment:
Are you sure the `Local.ROOT` local is the best here? IIRC, there are some languages that might not lower case some letters.

Why not let it `user.language` dependent ?

Reply:
hum you are right. I think we should go with Locale.English here, that should make sure it always gets lowercased correctly (or are there domain names not using a-z but another alphabet? i don't think so, but not 100% sure). Will fix :)

Reply:
so.. should we remove the actual validation check for now and just go with the lowercase approach to mitigate the actual issue? perhaps i overdid it a bit, wasnt aware of all the edge cases

Reply:
Hope now you see the challenge around hostname :)

For me, I would just give up on this change and update the documentation accordingly with a note/warning/tip.

Reply:
sounds like a plan. closing this for now and will create another PR with updated hostname guide.

Reply:
https://github.com/keycloak/keycloak/pull/11162